sections/main-blog.liquid

```diff
@@ -16,11 +16,21 @@
   }
 {%- endstyle -%}
 
+{%- liquid
+  if settings.spacing_grid_horizontal == 0 and settings.spacing_grid_vertical == 0
+    assign collapsed_class = 'blog--collapsed blog--collapsed-x blog--collapsed-y'
+  elsif settings.spacing_grid_horizontal == 0
+    assign collapsed_class = 'blog--collapsed blog--collapsed-x'
+  elsif settings.spacing_grid_vertical == 0
+    assign collapsed_class = 'blog--collapsed blog--collapsed-y'
+  endif
+-%}
```

Comment:
That's exactly what I described to Sofia as something I wasn't really excited to do. I suppose it's not _too_ bad without needing to also factor in the card type exception, but I think I'm fine keeping them copied in collage and blog. I'd like to see blog sharing grid styles in the future so I don't want to go out of our way too much for it. If we change our mind it can always be done in a cleanup PR.